01_homework/hw1_task3.cc

```diff
@@ -0,0 +1,102 @@
+#include <vector>
+#include <thread>
+#include <iterator>
+#include <numeric>
+#include <iostream>
+#include <algorithm>
+//#include <execution>
+
+template<typename T>
+using iter_type = typename std::iterator_traits<T>::difference_type;
+
+template<typename Iterator, typename Predicate, typename T>
+struct count_if_block
+{
+    void operator()(Iterator first, Iterator last, Predicate &policy, T& result)
```

Comment:
Nie podoba mi się, że zmiennej typu Predicate dałeś nazwę policy. To może być mylące, bo to policy kojarzy się właśnie z rodzajem std::execution.

Reply:
Zmieniłem nazwę na condition